src/sql/workbench/services/accountManagement/test/browser/accountManagementService.test.ts

```diff
@@ -421,13 +430,25 @@ suite('Account Management Service Tests:', () => {
 
 		// ... Add mocking for instantiating an account dialog controller
 		let mockDialogController = TypeMoq.Mock.ofType(AccountDialogController);
+		let mockAccountDialog = {};
 		mockDialogController.setup(x => x.openAccountDialog());
+		mockDialogController.setup(x => x.accountDialog).returns(() => <AccountDialog>mockAccountDialog);
+		let mockAccountDialogCloseEvent = new Emitter<void>();
+		mockAccountDialog['onCloseEvent'] = mockAccountDialogCloseEvent.event;
+		setTimeout(() => {
+			mockAccountDialogCloseEvent.fire();
+		}, 1000);
 		state.instantiationService.setup(x => x.createInstance(TypeMoq.It.isValue(AccountDialogController)))
 			.returns(() => mockDialogController.object);
 
 		// If: I open the account dialog for a second time
 		return state.accountManagementService.openAccountListDialog()
```

Comment:
So this test is actually broken as it is - since it doesn't use a done callback it will just completely immediately once the promise is returned instead of waiting for the verifies to be called. 

It'd be nice to fix this now - but if we do so that would be in a separate PR I'd say.

Reply:
mocha works with Promise, if you return a promise, it will wait for the promise to be fulfilled:
https://mochajs.org/#working-with-promises

Reply:
Oh right. Although I'm not sure what the intention is with firing the dialog close event. And why is that on a timer? That seems like a possibility of a race condition if you're relying on that event to trigger some state to change because of that...

Reply:
since we are waiting for the account dialog to be closed now, we have to mock the dialog close event, otherwise the test will fail with timeout error. I am trying to fire the event after the controller.openAccountDialog() is called, below is the accountManagementService's implementation, all of the objects involved are mocked and the execution shouldn't take much time, 1 sec is more than enough for now without needing to introduce more complicated mechanism.
`
	public openAccountListDialog(): Thenable<void> {
		let self = this;

		return new Promise((resolve, reject) => {
			try {
				// If the account list dialog hasn't been defined, create a new one
				if (!self._accountDialogController) {
					self._accountDialogController = self._instantiationService.createInstance(AccountDialogController);
				}
				self._accountDialogController.openAccountDialog();
				self._accountDialogController.accountDialog.onCloseEvent(resolve);
			} catch (e) {
				reject(e);
			}
		});
	}
`